File: all-submissions/0776-n-ary-tree-postorder-traversal/solution.java
/*
// Definition for a Node.
class Node {
    public int val;
    public List<Node> children;

    public Node() {}

    public Node(int _val,List<Node> _children) {
        val = _val;
        children = _children;
    }
};
*/
class Solution {
   public List<Integer> postorder(Node root) {
        List<Integer> ans = new ArrayList<>();
        if(root==null)
            return ans;
        
        for(Node i : root.children)
        {
            ans.addAll(postorder(i));
        }
       ans.add(root.val);
        return ans;
    }
}

File: all-submissions/0776-n-ary-tree-postorder-traversal/solution.py
"""
# Definition for a Node.
class Node:
    def __init__(self, val: Optional[int] = None, children: Optional[List['Node']] = None):
        self.val = val
        self.children = children
"""

class Solution:
    def postorder(self, root: 'Node') -> List[int]:
        if not root: return []
        res = []
        
        def postorderrec(node: 'Node'):
            if not node:
                return
            if not node.children:
                return
            for child in node.children:
                postorderrec(child)
                res.append(child.val)
        
        postorderrec(root)
        res.append(root.val)
        return res
        
        
